src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/FallbackProjectManager.cs

```diff
@@ -114,11 +114,18 @@ private async Task AddFallbackProjectAsync(ProjectId projectId, string filePath,
         // the project will be updated, and it will no longer be a fallback project.
         var hostProject = new FallbackHostProject(project.FilePath, intermediateOutputPath, FallbackRazorConfiguration.Latest, rootNamespace, project.Name);
 
-        await _dispatcher
-            .RunOnDispatcherThreadAsync(
-                () => _projectManagerAccessor.Instance.ProjectAdded(hostProject),
-                cancellationToken)
-            .ConfigureAwait(false);
+        if (_dispatcher.IsDispatcherThread)
```

Comment:
Should this logic just be moved into `RunOnDispatcherThread` ?

Reply:
100%! But I'm just making a small change for the 17.10p2 QB. I'll have a proper PR that updates dispatcher on main up shortly.

Reply:
Works for me